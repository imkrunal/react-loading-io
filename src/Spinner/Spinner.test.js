import React from 'react'
import serializer from 'jest-emotion'
import renderer from 'react-test-renderer'
import { Spinner } from '..'

expect.addSnapshotSerializer(serializer)

it('spinner renders correctly', () => {
  const tree = renderer.create(<Spinner />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('px converts', () => {
  const tree = renderer.create(<Spinner size={10} sizeUnit="em" />).toJSON()
  expect(tree).toMatchSnapshot()
})
