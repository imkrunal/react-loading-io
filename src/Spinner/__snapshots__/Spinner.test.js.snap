// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`px converts 1`] = `
@keyframes animation-0 {
  0% {
    opacity: 1;
  }

  100% {
    opacity: 0;
  }
}

.emotion-0 {
  left: 94px;
  top: 48px;
  position: absolute;
  -webkit-animation: animation-0 linear 1s infinite;
  animation: animation-0 linear 1s infinite;
  background: #f08d43;
  width: 12px;
  height: 24px;
  border-radius: 40%;
  -webkit-transform-origin: 6px 52px;
  -ms-transform-origin: 6px 52px;
  transform-origin: 6px 52px;
}

.emotion-0:nth-of-type(1) {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-animation-delay: 0.9166666666666666s;
  animation-delay: 0.9166666666666666s;
}

.emotion-0:nth-of-type(2) {
  -webkit-transform: rotate(30deg);
  -ms-transform: rotate(30deg);
  transform: rotate(30deg);
  -webkit-animation-delay: 0.8333333333333334s;
  animation-delay: 0.8333333333333334s;
}

.emotion-0:nth-of-type(3) {
  -webkit-transform: rotate(60deg);
  -ms-transform: rotate(60deg);
  transform: rotate(60deg);
  -webkit-animation-delay: 0.75s;
  animation-delay: 0.75s;
}

.emotion-0:nth-of-type(4) {
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  -webkit-animation-delay: 0.6666666666666666s;
  animation-delay: 0.6666666666666666s;
}

.emotion-0:nth-of-type(5) {
  -webkit-transform: rotate(120deg);
  -ms-transform: rotate(120deg);
  transform: rotate(120deg);
  -webkit-animation-delay: 0.5833333333333334s;
  animation-delay: 0.5833333333333334s;
}

.emotion-0:nth-of-type(6) {
  -webkit-transform: rotate(150deg);
  -ms-transform: rotate(150deg);
  transform: rotate(150deg);
  -webkit-animation-delay: 0.5s;
  animation-delay: 0.5s;
}

.emotion-0:nth-of-type(7) {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
  -webkit-animation-delay: 0.4166666666666667s;
  animation-delay: 0.4166666666666667s;
}

.emotion-0:nth-of-type(8) {
  -webkit-transform: rotate(210deg);
  -ms-transform: rotate(210deg);
  transform: rotate(210deg);
  -webkit-animation-delay: 0.3333333333333333s;
  animation-delay: 0.3333333333333333s;
}

.emotion-0:nth-of-type(9) {
  -webkit-transform: rotate(240deg);
  -ms-transform: rotate(240deg);
  transform: rotate(240deg);
  -webkit-animation-delay: 0.25s;
  animation-delay: 0.25s;
}

.emotion-0:nth-of-type(10) {
  -webkit-transform: rotate(270deg);
  -ms-transform: rotate(270deg);
  transform: rotate(270deg);
  -webkit-animation-delay: 0.16666666666666666s;
  animation-delay: 0.16666666666666666s;
}

.emotion-0:nth-of-type(11) {
  -webkit-transform: rotate(300deg);
  -ms-transform: rotate(300deg);
  transform: rotate(300deg);
  -webkit-animation-delay: 0.08333333333333333s;
  animation-delay: 0.08333333333333333s;
}

.emotion-0:nth-of-type(12) {
  -webkit-transform: rotate(330deg);
  -ms-transform: rotate(330deg);
  transform: rotate(330deg);
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
}

.emotion-12 {
  position: relative;
  width: 10em !important;
  height: 10em !important;
  -webkit-transform: translate( -8em,-8em ) scale(0.8) translate( 8em,8em );
  -ms-transform: translate( -8em,-8em ) scale(0.8) translate( 8em,8em );
  transform: translate( -8em,-8em ) scale(0.8) translate( 8em,8em );
}

<div>
  <div
    className="emotion-12"
  >
    <div
      className="emotion-0"
    />
    <div
      className="emotion-0"
    />
    <div
      className="emotion-0"
    />
    <div
      className="emotion-0"
    />
    <div
      className="emotion-0"
    />
    <div
      className="emotion-0"
    />
    <div
      className="emotion-0"
    />
    <div
      className="emotion-0"
    />
    <div
      className="emotion-0"
    />
    <div
      className="emotion-0"
    />
    <div
      className="emotion-0"
    />
    <div
      className="emotion-0"
    />
  </div>
</div>
`;

exports[`spinner renders correctly 1`] = `
@keyframes animation-0 {
  0% {
    opacity: 1;
  }

  100% {
    opacity: 0;
  }
}

.emotion-12 {
  position: relative;
  width: 200px !important;
  height: 200px !important;
  -webkit-transform: translate( -200px,-200px ) scale(1) translate( 200px,200px );
  -ms-transform: translate( -200px,-200px ) scale(1) translate( 200px,200px );
  transform: translate( -200px,-200px ) scale(1) translate( 200px,200px );
}

.emotion-0 {
  left: 94px;
  top: 48px;
  position: absolute;
  -webkit-animation: animation-0 linear 1s infinite;
  animation: animation-0 linear 1s infinite;
  background: #f08d43;
  width: 12px;
  height: 24px;
  border-radius: 40%;
  -webkit-transform-origin: 6px 52px;
  -ms-transform-origin: 6px 52px;
  transform-origin: 6px 52px;
}

.emotion-0:nth-of-type(1) {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-animation-delay: 0.9166666666666666s;
  animation-delay: 0.9166666666666666s;
}

.emotion-0:nth-of-type(2) {
  -webkit-transform: rotate(30deg);
  -ms-transform: rotate(30deg);
  transform: rotate(30deg);
  -webkit-animation-delay: 0.8333333333333334s;
  animation-delay: 0.8333333333333334s;
}

.emotion-0:nth-of-type(3) {
  -webkit-transform: rotate(60deg);
  -ms-transform: rotate(60deg);
  transform: rotate(60deg);
  -webkit-animation-delay: 0.75s;
  animation-delay: 0.75s;
}

.emotion-0:nth-of-type(4) {
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  -webkit-animation-delay: 0.6666666666666666s;
  animation-delay: 0.6666666666666666s;
}

.emotion-0:nth-of-type(5) {
  -webkit-transform: rotate(120deg);
  -ms-transform: rotate(120deg);
  transform: rotate(120deg);
  -webkit-animation-delay: 0.5833333333333334s;
  animation-delay: 0.5833333333333334s;
}

.emotion-0:nth-of-type(6) {
  -webkit-transform: rotate(150deg);
  -ms-transform: rotate(150deg);
  transform: rotate(150deg);
  -webkit-animation-delay: 0.5s;
  animation-delay: 0.5s;
}

.emotion-0:nth-of-type(7) {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
  -webkit-animation-delay: 0.4166666666666667s;
  animation-delay: 0.4166666666666667s;
}

.emotion-0:nth-of-type(8) {
  -webkit-transform: rotate(210deg);
  -ms-transform: rotate(210deg);
  transform: rotate(210deg);
  -webkit-animation-delay: 0.3333333333333333s;
  animation-delay: 0.3333333333333333s;
}

.emotion-0:nth-of-type(9) {
  -webkit-transform: rotate(240deg);
  -ms-transform: rotate(240deg);
  transform: rotate(240deg);
  -webkit-animation-delay: 0.25s;
  animation-delay: 0.25s;
}

.emotion-0:nth-of-type(10) {
  -webkit-transform: rotate(270deg);
  -ms-transform: rotate(270deg);
  transform: rotate(270deg);
  -webkit-animation-delay: 0.16666666666666666s;
  animation-delay: 0.16666666666666666s;
}

.emotion-0:nth-of-type(11) {
  -webkit-transform: rotate(300deg);
  -ms-transform: rotate(300deg);
  transform: rotate(300deg);
  -webkit-animation-delay: 0.08333333333333333s;
  animation-delay: 0.08333333333333333s;
}

.emotion-0:nth-of-type(12) {
  -webkit-transform: rotate(330deg);
  -ms-transform: rotate(330deg);
  transform: rotate(330deg);
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
}

<div>
  <div
    className="emotion-12"
  >
    <div
      className="emotion-0"
    />
    <div
      className="emotion-0"
    />
    <div
      className="emotion-0"
    />
    <div
      className="emotion-0"
    />
    <div
      className="emotion-0"
    />
    <div
      className="emotion-0"
    />
    <div
      className="emotion-0"
    />
    <div
      className="emotion-0"
    />
    <div
      className="emotion-0"
    />
    <div
      className="emotion-0"
    />
    <div
      className="emotion-0"
    />
    <div
      className="emotion-0"
    />
  </div>
</div>
`;
